specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2020-01-01-preview/managedcluster.json

```diff
@@ -0,0 +1,1216 @@
+      "required": [
+        "dnsName",
+        "adminUserName"
+      ],
+      "properties": {
+        "dnsName": {
+          "type": "string",
+          "description": "The cluster dns name."
+        },
+        "fqdn": {
+          "type": "string",
+          "description": "the cluster Fully qualified domain name.",
+          "readOnly": true
+        },
+        "clusterId": {
+          "type": "string",
+          "description": "A service generated unique identifier for the cluster resource.",
+          "readOnly": true
+        },
+        "clusterState": {
+          "readOnly": true,
+          "$ref": "#/definitions/ClusterState",
+          "description": "The current state of the cluster.\n\n  - WaitingForNodes - Indicates that the cluster resource is created and the resource provider is waiting for Service Fabric VM extension to boot up and report to it.\n  - Deploying - Indicates that the Service Fabric runtime is being installed on the VMs. Cluster resource will be in this state until the cluster boots up and system services are up.\n  - BaselineUpgrade - Indicates that the cluster is upgrading to establishes the cluster version. This upgrade is automatically initiated when the cluster boots up for the first time.\n  - UpdatingUserConfiguration - Indicates that the cluster is being upgraded with the user provided configuration.\n  - UpdatingUserCertificate - Indicates that the cluster is being upgraded with the user provided certificate.\n  - UpdatingInfrastructure - Indicates that the cluster is being upgraded with the latest Service Fabric runtime version. This happens only when the **upgradeMode** is set to 'Automatic'.\n  - EnforcingClusterVersion - Indicates that cluster is on a different version than expected and the cluster is being upgraded to the expected version.\n  - UpgradeServiceUnreachable - Indicates that the system service in the cluster is no longer polling the Resource Provider. Clusters in this state cannot be managed by the Resource Provider.\n  - AutoScale - Indicates that the ReliabilityLevel of the cluster is being adjusted.\n  - Ready - Indicates that the cluster is in a stable state.\n"
+        },
+        "clusterCertificateThumbprint": {
+          "type": "string",
+          "description": "The cluster certificate thumbprint used node to node communication.",
+          "readOnly": true
+        },
+        "clientConnectionPort": {
+          "type": "integer",
+          "description": "The port used for client connections to the cluster.",
+          "default": 19000
+        },
+        "httpGatewayConnectionPort": {
+          "type": "integer",
+          "description": "The port used for http connections to the cluster.",
+          "default": 19080
+        },
+        "adminUserName": {
+          "type": "string",
+          "description": "vm admin user name."
+        },
+        "adminPassword": {
+          "type": "string",
+          "format": "password",
```

Comment:
>          "format": "password [](start = 0, length = 29)

Please go through [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=best%20practice](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=best%20practice)
In this case please annotate using x-ms-secret: true

Reply:
done thanks